### DISCUSSION TITLE
Entity caching

### DISCUSSION THREAD

**Discussion by Geal** [apollographql/router#4592] [Preview]
This is the place to discuss the upcoming preview feature Entity caching

**Comment by smyrick** [comment]
We have a customer today who asked if the Entity cache will support different cache policies and strategies similar to the Apollo Client cache.

The want to have cached data live for a certain time, but then at expiration or as determined by them they want to try fetching data from a subgraph but if that fails, fallback to returning cache data.

This got me thinking more generally about different cache policies we could have for the Router. This could allow us to have a circuit breaker style implementation where we cache until we find the service is back online.
